syft/generic/object_storage.py

```diff
@@ -126,3 +133,28 @@ def clear_objects(self, return_self: bool = True):
     def current_objects(self):
         """Returns a copy of the objects in the object storage."""
         return self._objects.copy()
+
+    def find_by_id(self, id):
+        """Local search by id"""
+        if id in self._objects:
+            return self._objects[id]
+        else:
+            return None
+
+    def find_by_tag(self, tag):
+        """Local search by tag
+
+        Args:
+            tag (str): exact tag searched
+
+        Return:
+            A list of results, possibly empty
+        """
+        if tag in self._tag_to_object_ids:
+            results = []
```

Comment:
I may like list comprehension too much but maybe you can use one here (if it makes the code clearer)

Reply:
hum not sure it wouldn't keep on a single line. Unless you have an idea?

Reply:
```result = [
    self.find_by_id(obj_id)
    for obj_id in self._tag_to_object_ids[tag]
    if self.find_by_id(obj_id) is not None
]
```
But I'm not sure if it's better because I have find_by_id twice. You can keep it as is :)

Reply:
Will do :)